src/libcore/slice/mod.rs

```diff
@@ -2145,6 +2146,29 @@ impl<T> [T] {
         }
     }
 
+    /// Fills `self` with elements by cloning `value`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_fill)]
+    ///
+    /// let mut buf = vec![0; 10];
+    /// buf.fill(1);
+    /// assert_eq!(buf, vec![1; 10]);
+    /// ```
+    #[unstable(feature = "slice_fill", issue = "70758")]
+    pub fn fill<V>(&mut self, value: V)
+    where
+        V: Borrow<T>,
+        T: Clone,
+    {
+        let value = value.borrow();
+        for el in self {
+            *el = value.clone();
```

Comment:
```suggestion
            el.clone_from(value);
```

Reply:
Oh cool, I didn't know about [`Clone::clone_from`](https://doc.rust-lang.org/std/clone/trait.Clone.html#method.clone_from). That indeed seems like a nice improvement. Thanks for suggesting!